metashare/api/jobs.py

```diff
@@ -0,0 +1,178 @@
+    # are done.
+
+    return org_config
+
+
+def try_to_make_branch(repository, *, new_branch, base_branch):
+    branch_name = new_branch
+    counter = 0
+    while True:
+        try:
+            latest_sha = repository.branch(base_branch).latest_sha()
+            repository.create_branch_ref(branch_name, latest_sha)
+            return branch_name
+        except UnprocessableEntity:
+            counter += 1
+            branch_name = f"{new_branch}-{counter}"
+
+
+def create_branches_on_github(*, user, repo_url, project, task, repo_root):
+    """
+    Expects to be called in the context of a local github checkout.
+    """
+    gh = login(token=user.gh_token)
+    owner, repo = extract_owner_and_repo(repo_url)
+    repository = gh.repository(owner, repo)
+
+    # Make project branch:
+    if project.branch_name:
+        project_branch_name = project.branch_name
+    else:
+        prefix = get_cumulus_prefix(
+            repo_root=repo_root,
+            repo_name="Repo Name",  # A placeholder, not used.
+            repo_url=repo_url,
+            repo_owner=owner,
+            repo_branch=repository.default_branch,
+            repo_commit=repository.branch(repository.default_branch).latest_sha(),
+        )
+        project_branch_name = f"{prefix}{slugify(project.name)}"
+        project_branch_name = try_to_make_branch(
+            repository,
+            new_branch=project_branch_name,
+            base_branch=repository.default_branch,
+        )
+        project.branch_name = project_branch_name
+
+    # Make task branch:
+    if task.branch_name:
+        task_branch_name = task.branch_name
+    else:
+        task_branch_name = try_to_make_branch(
+            repository,
+            new_branch=f"{project_branch_name}__{slugify(task.name)}",
+            base_branch=project_branch_name,
+        )
+        task.branch_name = task_branch_name
+
+    project.save()
+    task.save()
+
+    return task_branch_name
+
+
+def run_appropriate_flow(scratch_org, *, user, repo_url, repo_branch):
+    from .models import SCRATCH_ORG_TYPES
+
+    cases = {SCRATCH_ORG_TYPES.Dev: "dev_org", SCRATCH_ORG_TYPES.QA: "qa_org"}
+
+    owner, repo = extract_owner_and_repo(repo_url)
+    run_flow(
+        repo_owner=owner,
+        repo_name=repo,
+        repo_branch=repo_branch,
+        user=user,
+        flow_name=cases[scratch_org.org_type],
+    )
+
+
+def create_branches_on_github_then_create_scratch_org(
+    *, project, repo_url, scratch_org, task, user
+):
+    with contextlib.ExitStack() as stack:
+        repo_root = stack.enter_context(local_github_checkout(user, repo_url))
+        stack.enter_context(report_errors_on(scratch_org))
+
+        commit_ish = create_branches_on_github(
+            user=user,
+            repo_url=repo_url,
+            project=project,
+            task=task,
+            repo_root=repo_root,
+        )
+        create_scratch_org(
```

Comment:
Return org here, once it's made in sf_run_flows style. Then pass it in to run_appropriate_flow, which will then be redesigned to presume it gets passed the org and doesn't have to make it.

Reply:
Instead, I'm using the combined create-org and run-flow logic in `sf_run_flows`.